### PR TITLE
fix(evm): Receipt hashes are incorrect.

### PR DESCRIPTION
**Pull request by itsdevbear**
Discovered via testing OP-stack deployments, that our receipt hashes were incorrect.

This PR fixes it.

### CODE REVIEW

**Review by oofncubed — APPROVED** [review]
utACK

**Comment by itsdevbear** [comment]
@calbera gonna be busy with meetings today, can you move the log information (apart from block hash) back into the StateDB to match Geth. 

Add some additional tests as well.

File: eth/core/processor.go [hunk 1]
```diff
@@ -161,40 +164,52 @@ func (sp *StateProcessor) ProcessTransaction(
...
-	if result.Failed() {
-		receipt.Status = types.ReceiptStatusFailed
-	} else {
-		receipt.Status = types.ReceiptStatusSuccessful
+	// Add the logs, with block metadata, to the receipt; the block hash has not been computed
+	// since the block is not complete at this point.
+	receipt.Logs = sp.statedb.GetLogs(receipt.TxHash, sp.header.Number.Uint64(), common.Hash{})
+	for _, log := range receipt.Logs {
+		log.Index = sp.logIndex
+		sp.logIndex++
```
Comment (itsdevbear) [inline eth/core/processor.go:207]: @calbera log index can go on the logs journal

File: eth/core/imported.go [hunk 2]
```diff
@@ -56,6 +56,9 @@ var (
 	GetHashFn = core.GetHashFn
 	// TransactionToMessage converts a transaction to a message.
 	TransactionToMessage = core.TransactionToMessage
+
+	Transfer    = core.Transfer
```
Comment (itsdevbear) [inline eth/core/imported.go:60]: @calbera why imported?

File: eth/core/processor_test.go [hunk 3]
```diff
@@ -44,73 +44,52 @@ var (
...
-		) {
-			// no-op
-		}
-
-		gp.SetBlockGasLimit(blockGasLimit)
-		sp.Prepare(context.Background(), nil, dummyHeader)
+		sp = core.NewStateProcessor(cp, gp, pp, sdb, &vm.Config{})
+		Expect(sp).ToNot(BeNil())
+		evm = vm.NewGethEVMWithPrecompiles(
+			vm.BlockContext{
```
Comment (itsdevbear) [inline eth/core/processor_test.go:87]: don't need to do this, there is a function that'll build you a context, saves the Transfer and CanTransfer imports.

**Comment by itsdevbear** [comment]
@calbera good with this?

**Comment by itsdevbear** [comment]
Oh wait we gotta fix that bug Diego had

**Pull request merged**